File: airesources/Java/Networking.java
package halitejavabot;

import java.io.FileWriter;
import java.io.IOException;
import java.util.Arrays;
import java.util.LinkedList;
import java.util.Vector;

public class Networking {
    private static final char NEW_LINE = '\n';
    private static final char CARRIAGE_RETURN = '\r';
    private static final char UNDOCK_KEY = 'u';
    private static final char DOCK_KEY = 'd';
    private static final char THRUST_KEY = 't';
    private static final String SPACE = " ";

    public static void sendMoves(Vector<Move> moves) {
        StringBuilder moveString = new StringBuilder();
        for (Move move : moves) {
            switch (move.getType()) {
                case Noop:
                    continue;
                case Undock:
                    moveString.append(UNDOCK_KEY)
                            .append(SPACE)
                            .append(move.getShip().getId().getId())
                            .append(SPACE);
                    break;
                case Dock:
                    moveString.append(DOCK_KEY)
                            .append(SPACE)
                            .append(move.getShip().getId().getId())
                            .append(SPACE)
                            .append(((DockMove) move).getDestination().getId())
                            .append(SPACE);
                    break;
                case Thrust:
                    moveString.append(THRUST_KEY)
                            .append(SPACE)
                            .append(move.getShip().getId().getId())
                            .append(SPACE)
                            .append(((ThrustMove) move).getThrust())
                            .append(SPACE)
                            .append(((ThrustMove) move).getAngle())
                            .append(SPACE);
                    break;
            }
        }
        System.out.println(moveString);
    }

    private static String getString() {
        try {
            StringBuilder builder = new StringBuilder();
            int buffer;
            while ((buffer = System.in.read()) >= 0) {
                if (buffer == NEW_LINE) {
                    break;
                } else {
                    builder = builder.append((char)buffer);
                }
            }
            if(builder.charAt(builder.length() - 1) == CARRIAGE_RETURN)
                builder.setLength(builder.length() - 1); //Removes a carriage return if on windows for manual testing.
            return builder.toString();
        } catch(Exception e) {
            System.exit(1);
            return null;
        }
    }

    static LinkedList<String> parseInput() {
        return new LinkedList<>(Arrays.asList(getString().trim().split(SPACE)));
    }

    GameMap initialize(String botName) {
        short myId = Short.parseShort(getString());

        try {
            DebugLog.initialize(new FileWriter(String.format("%d - %s.log", myId, botName)));
        } catch (IOException e) {
            e.printStackTrace();
        }

        LinkedList<String> inputStringComponents = parseInput();

        short width = Short.parseShort(inputStringComponents.pop());
        short height = Short.parseShort(inputStringComponents.pop());

        GameMap gameMap = new GameMap(width, height, myId);

        // Associate bot name
        System.out.println(botName);

        inputStringComponents = Networking.parseInput();
        gameMap.updateMap(inputStringComponents);

        // Initialize debugging log

        return gameMap;
    }

}

File: airesources/CSharp/Halite2/hlt/Networking.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace Halite2.hlt
{
    public class Networking
    {

        private static char UNDOCK_KEY = 'u';
        private static char DOCK_KEY = 'd';
        private static char THRUST_KEY = 't';

        public static void SendMoves(IEnumerable<Move> moves)
        {
            StringBuilder moveString = new StringBuilder();

            foreach (Move move in moves)
            {
                switch (move.GetMoveType())
                {
                    case Move.MoveType.Noop:
                        continue;
                    case Move.MoveType.Undock:
                        moveString.Append(UNDOCK_KEY)
                                .Append(" ")
                                .Append(move.GetShip().GetId())
                                .Append(" ");
                        break;
                    case Move.MoveType.Dock:
                        moveString.Append(DOCK_KEY)
                                .Append(" ")
                                .Append(move.GetShip().GetId())
                                .Append(" ")
                                .Append(((DockMove)move).GetDestinationId())
                                .Append(" ");
                        break;
                    case Move.MoveType.Thrust:
                        moveString.Append(THRUST_KEY)
                                .Append(" ")
                                .Append(move.GetShip().GetId())
                                .Append(" ")
                                .Append(((ThrustMove)move).GetThrust())
                                .Append(" ")
                                .Append(((ThrustMove)move).GetAngle())
                                .Append(" ");
                        break;
                }
            }
            Console.WriteLine(moveString);
        }

        private static String ReadLine()
        {
            try
            {
                StringBuilder builder = new StringBuilder();
                int buffer;

                for (; (buffer = Console.Read()) >= 0;) {
                    if (buffer == '\n')
                    {
                        break;
                    }
                    if (buffer == '\r')
                    {
                        // Ignore carriage return if on windows for manual testing.
                        continue;
                    }
                    builder = builder.Append((char)buffer);
                }
                return builder.ToString();
            }
            catch (Exception)
            {
                Environment.Exit(0);
                return null;
            }
        }

        public static Metadata ReadLineIntoMetadata()
        {
            return new Metadata(ReadLine().Trim().Split(' '));
        }

        public GameMap Initialize(String botName)
        {
            int myId = int.Parse(ReadLine());
            Log.Initialize(new StreamWriter(String.Format("{0}_{1}.log", myId, botName)));

            Metadata inputStringMapSize = ReadLineIntoMetadata();
            int width = int.Parse(inputStringMapSize.Pop());
            int height = int.Parse(inputStringMapSize.Pop());
            GameMap gameMap = new GameMap(width, height, myId);

            // Associate bot name
            Console.WriteLine(botName);

            Metadata inputStringMetadata = ReadLineIntoMetadata();
            gameMap.UpdateMap(inputStringMetadata);

            return gameMap;
        }
    }

}
